JPF/mysql.skeleton.ipynb

```diff
@@ -152,36 +156,38 @@
     "    return dtypedict\n",
     "\n",
     "\n",
-    "def hpd_csv2sql(description, input_csv_file, sep_char, output_pickle,\\\n",
+    "def hpd_csv2sql(description, input_csv_url, sep_char,\\\n",
     "            table_name, dtype_dict, load_pickle, \\\n",
-    "                input_pickle, db_action, truncate_columns, date_time_columns,\\\n",
+    "                pickle_file, db_action, truncate_columns, date_time_columns,\\\n",
     "               chunk_size, keep_cols, max_col_len=255):\n",
-    "\n",
+    "    \n",
     "    log.info(\"Beginning {} Import {}\".format(description,datetime.datetime.now()))\n",
     "    \n",
-    "    if os.path.isfile(input_pickle):\n",
-    "        log.info(\"Flagged load of PICKLE: {} = True\".format(input_pickle))\n",
+    "    if load_pickle: #IF FLAGGED TO LOAD PICKLE AS TRUE\n",
+    "        log.info(\"Flagged load of PICKLE: {} = True\".format(pickle_file))\n",
     "        \n",
-    "        with open(input_pickle, 'r') as picklefile:\n",
-    "            log.info(\"Begin OPEN {} Pickle: {}\".format(input_pickle, datetime.datetime.now()))\n",
-    "            log.info(\"Great we have a pickle file...Loading from {}\".format(input_pickle))\n",
+    "        with open(pickle_file, 'r') as picklefile:\n",
+    "            log.info(\"Begin OPEN {} Pickle: {}\".format(pickle_file, datetime.datetime.now()))\n",
+    "            log.info(\"Great we have a pickle file...Loading from {}\".format(pickle_file))\n",
     "            df = pickle.load(picklefile)\n",
     "\n",
     "    else: \n",
-    "        log.info(\"Reading CSV from {} .. This may take a while...\".format(input_csv_file))\n",
+    "        log.info(\"Reading CSV from {} .. This may take a while...\".format(input_csv_url))\n",
     "        \n",
-    "        with open(input_csv_file, 'r') as input_csv: ## should just change to IF EXISTS rather than open()???\n",
-    "            df = pd.read_csv(input_csv_file , sep=sep_char, dtype=dtype_dict)\n",
-    "                \n",
-    "        log.info(\"Why don't we save {} for next time\".format(output_pickle))\n",
+    "        #SWITCHING TO URL DIRECTLY, CAN'T CHECK FOR LOCAL FILE ANYMORE\n",
```

Comment:
I'd argue that we should try to maintain a workflow of download -> pickle -> load, with checks that will skip the download and pickle steps if on disk.

The biggest reason being that opening the file via network every time is pretty inefficient. Any issue in pickling or loading and you have re-download and re-pickle again. 

Also, while I'm not super familiar with pandas internals, opening via http probably doesn't give us the option to read the csv in chunks, resulting in a more memory efficient load process. For example, I've been watching this process for ~4 hours now.